helix-core/src/test/java/org/apache/helix/integration/paticipant/TestInstanceAutoJoin.java

```diff
@@ -110,14 +108,23 @@ public void testAutoRegistrationShouldFailWhenWaitingResponse() throws Exception
         new MockParticipantManager(ZK_ADDR, CLUSTER_NAME, instance3);
     autoParticipant.syncStart();
 
-    Assert.assertTrue(null == manager.getHelixDataAccessor()
-        .getProperty(accessor.keyBuilder().liveInstance(instance3)));
+    // if the test is run in cloud environment, auto registration will succeed and live instance
+    // will be added, otherwise, auto registration will fail and instance config will not be
+    // populated. An exception will be thrown.
     try {
       manager.getConfigAccessor().getInstanceConfig(CLUSTER_NAME, instance3);
```

Comment:
Why this line is needed?

Reply:
This line is used to trigger the exception to be thrown as the instance config is not there.

Reply:
Got it. You are right.